packages/block-library/src/pattern/index.php

```diff
@@ -22,14 +22,22 @@ function register_block_core_pattern() {
 /**
  * Renders the `core/pattern` block on the server.
  *
- * @param array $attributes Block attributes.
+ * @param array  $attributes Block attributes.
+ * @param string $content    The block rendered content.
  *
  * @return string Returns the output of the pattern.
  */
-function render_block_core_pattern( $attributes ) {
+function render_block_core_pattern( $attributes, $content ) {
 	if ( empty( $attributes['slug'] ) ) {
 		return '';
 	}
+	$slug_classname = str_replace( '/', '-', $attributes['slug'] );
+	$classnames     = isset( $attributes['className'] ) ? $attributes['className'] . ' ' . $slug_classname : $slug_classname;
+	$wrapper        = '<div class="' . esc_attr( $classnames ) . '">%s</div>';
+
+	if ( isset( $attributes['syncStatus'] ) && 'unsynced' === $attributes['syncStatus'] ) {
```

Comment:
It doesn't appear to be possible to ever have `unsynced` as a value:
<img width="496" alt="Screenshot 2023-05-15 at 10 28 40" src="https://github.com/WordPress/gutenberg/assets/699132/a4aafb61-3183-409a-98d9-094f9e71eae1">

Reply:
The thinking was that if there is no `syncStatus`, ie. `syncStatus===undefined` then the pattern is not synced, which is partly a backwards  compat approach as currently none of the patterns in the wild are synced and they have no `syncStatus` attrib. We could add another `syncStatus` of `unsynced` but we will then have to account for both that and `undefined` as an indicator of not synced.  

Adding `unsyncec` however may make it clearer to people browsing the block.json that there is an `unsynced` option and avoid the confusion you experienced. I don't have a strong opinion either way. @talldan, @aaronrobertshaw what do you think?

Reply:
Good catch @gziolo 👍 

> The thinking was that if there is no syncStatus, ie. syncStatus===undefined then the pattern is not synced

That's what the block.json attribute definition indicates. What I think @gziolo is questioning is the `index.php` code checking for an `unsynced` value. That appears to be a leftover from the [original PR](https://github.com/WordPress/gutenberg/pull/50272/files#diff-794801e5b887c97c685a20f7887edb0ea486e6c1fb4f14caf4f9c16cd1456dd1R37-R39) where `unsynced` was a valid `syncStatus` value and its default.

Reply:
🤦 missed the first code snippet in the original comment! Ignore all of my previous comment, I will get the `index.php` file sorted.

Reply:
[PR to fix this here](https://github.com/WordPress/gutenberg/pull/50646)